### PR TITLE
Migrate from test-framework to tasty

### PR DESCRIPTION
**Pull request by Bodigrim**
As discussed in #790.

### CODE REVIEW

File: containers-tests/tests/intmap-properties.hs [hunk 1]
```diff
@@ -1469,36 +1449,36 @@ prop_deleteMaxModel ys = length ys > 0 ==>
 prop_filter :: Fun Int Bool -> [(Int, Int)] -> Property
 prop_filter p ys = length ys > 0 ==>
   let xs = List.nubBy ((==) `on` fst) ys
-      m  = filter (apply p) (fromList xs)
+      m  = filter (applyFun p) (fromList xs)
```
Comment (treeowl) [inline containers-tests/tests/intmap-properties.hs:1452]: Why this change? `apply` isn't deprecated, and it's shorter.

Comment (Bodigrim) [reply containers-tests/tests/intmap-properties.hs]: No problem, let's stick to `apply`.

**Comment by treeowl** [comment]
Thanks!
